azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/NetworkInterfaceImpl.java

```diff
@@ -0,0 +1,407 @@
+        SubnetInner subnetInner = subnetToAssociate();
+        if (subnetInner != null) {
+            this.primaryIpConfiguration().setSubnet(subnetInner);
+        }
+
+        PublicIPAddressInner publicIpInner = publicIpToAssociate();
+        if (publicIpInner != null) {
+            this.primaryIpConfiguration().setPublicIPAddress(publicIpInner);
+        }
+
+        ServiceResponse<NetworkInterfaceInner> response = this.client.createOrUpdate(this.resourceGroupName(), this.nicName, this.inner());
+        this.setInner(response.getBody());
+    }
+
+    /**************************************************.
+     * Helper methods
+     **************************************************/
+
+    private void initialize() {
+        if (this.inner().ipConfigurations() == null) {
+            this.inner().setIpConfigurations(new ArrayList<NetworkInterfaceIPConfiguration>());
+        }
+
+        if (this.inner().dnsSettings() == null) {
+            this.inner().setDnsSettings(new NetworkInterfaceDnsSettings());
+        }
+    }
+
+    private String nameWithPrefix(String prefix) {
+        return prefix + "-" + this.uniqueId + "-" + this.resourceGroupName();
+    }
+
+    private NetworkInterfaceIPConfiguration primaryIpConfiguration() {
+        if (this.inner().ipConfigurations().size() == 0) {
+            this.inner().ipConfigurations().add(new NetworkInterfaceIPConfiguration());
+        }
+        return this.inner().ipConfigurations().get(0);
+    }
+
+    private List<String> dnsServerIps() {
+        if (this.inner().dnsSettings().dnsServers() == null) {
+            this.inner().dnsSettings().setDnsServers(new ArrayList<String>());
+        }
+        return this.inner().dnsSettings().dnsServers();
+    }
+
+    private PublicIpAddress.DefinitionCreatable prepareCreatablePublicIp(String name, String leafDnsLabel) {
+        PublicIpAddress.DefinitionWithGroup definitionWithGroup = this.publicIpAddresses
+                .define(name)
+                .withRegion(this.region());
+
+        PublicIpAddress.DefinitionAfterGroup definitionAfterGroup;
+        if (this.newGroup != null) {
+            definitionAfterGroup = definitionWithGroup.withNewGroup(this.newGroup);
+        } else {
+            definitionAfterGroup = definitionWithGroup.withExistingGroup(this.resourceGroupName());
+        }
+        return definitionAfterGroup.withLeafDomainLabel(leafDnsLabel);
+    }
+
+    private SubnetInner subnetToAssociate() {
+        SubnetInner subnetInner = new SubnetInner();
+        if (isInCreateMode()) {
+            // define..create mode
+            if (this.creatableVirtualNetworkKey != null) {
+                Network network = (Network) createdResource(this.creatableVirtualNetworkKey);
+                subnetInner.setId(network.subnets().get(0).inner().id());
+                return subnetInner;
+            }
+
+            for (Subnet subnet : this.existingVirtualNetworkToAssociate.subnets()) {
```

Comment:
although there is a return earlier, it may help for clarity to have explicit "else" blocks in such patterns, especially since the logic here is kind of special.